Flaky `BackendEstimator` test in CI

### Environment

- **Qiskit Terra version**: main @ pre-0.22rc1
- **Python version**: 3.10
- **Operating system**: Linux


### What is happening?

The `TestBackendEstimator.test_with_shots_option` test is flaky in CI; the tolerance of the test appears to be too low for the number of shots.  However, the test sets a seed and is still flaky.  This suggests that some part of the interface is not respecting the random seed.

### How can we reproduce the issue?

CI has been occasionally failing with:
```text
test.python.primitives.test_backend_estimator.TestBackendEstimator.test_run_with_shots_option_2
-----------------------------------------------------------------------------------------------

Captured traceback:
~~~~~~~~~~~~~~~~~~~
    Traceback (most recent call last):

      File "/home/vsts/work/1/s/test-job/lib/python3.10/site-packages/ddt.py", line 220, in wrapper
    return func(self, *args, **kwargs)

      File "/tmp/terra-tests/test/python/primitives/test_backend_estimator.py", line 231, in test_run_with_shots_option
    np.testing.assert_allclose(result.values, [-1.307397243478641], rtol=0.05)

      File "/home/vsts/work/1/s/test-job/lib/python3.10/site-packages/numpy/testing/_private/utils.py", line 1527, in assert_allclose
    assert_array_compare(compare, actual, desired, err_msg=str(err_msg),

      File "/home/vsts/work/1/s/test-job/lib/python3.10/site-packages/numpy/testing/_private/utils.py", line 844, in assert_array_compare
    raise AssertionError(msg)

    AssertionError: 
Not equal to tolerance rtol=0.05, atol=0

Mismatched elements: 1 / 1 (100%)
Max absolute difference: 0.07152427
Max relative difference: 0.05470737
 x: array([-1.235873])
 y: array([-1.307397])
```

### What should happen?

Reliable tests.

### Any suggestions?

In the immediate term, we can raise the tolerance of the test.  I ran this same test until failure 3 times, and got three failures with rtols of (0.0504, 0.0532, 0.0503) in 116 total tests.  So that's a failure rate of ~0.026.  If we assume (very hand-wavy) that the relative difference is normally distributed about 0, then this is approximately consistent with a standard deviation of 0.023.  If so, setting an rtol of 0.1 should give a failure rate of ~1e-5.

--- a/test/python/primitives/test_backend_estimator.py
+++ b/test/python/primitives/test_backend_estimator.py
@@ -225,10 +225,10 @@ class TestBackendEstimator(QiskitTestCase):
             [self.observable],
             parameter_values=[[0, 1, 1, 2, 3, 5]],
             shots=1024,
-            seed=15,
+            seed_simulator=15,
         ).result()
         self.assertIsInstance(result, EstimatorResult)
-        np.testing.assert_allclose(result.values, [-1.307397243478641], rtol=0.05)
+        np.testing.assert_allclose(result.values, [-1.307397243478641], rtol=0.1)
 
     @combine(backend=BACKENDS)
     def test_options(self, backend):
@@ -237,9 +237,9 @@ class TestBackendEstimator(QiskitTestCase):
             estimator = BackendEstimator(backend=backend, options={"shots": 3000})
             self.assertEqual(estimator.options.get("shots"), 3000)
         with self.subTest("set_options"):
-            estimator.set_options(shots=1024, seed=15)
+            estimator.set_options(shots=1024, seed_simulator=15)
             self.assertEqual(estimator.options.get("shots"), 1024)
-            self.assertEqual(estimator.options.get("seed"), 15)
+            self.assertEqual(estimator.options.get("seed_simulator"), 15)
         with self.subTest("run"):
             result = estimator.run(
                 [self.ansatz],

